File: 7-ReverseInteger/7-ReverseInteger.js
// Last updated: 3/18/2026, 11:03:11 PM
1/**
2 * @param {number} x
3 * @return {number}
4 */
5var reverse = function(x) {
6    let rev = 0;
7    let sign = x < 0 ? -1 : 1;
8    x = Math.abs(x);
9
10    while (x > 0) {
11        let digit = x % 10;
12        x = Math.floor(x / 10);
13
14        // Check for overflow before adding digit
15        if (rev > Math.floor((2**31 - 1) / 10)) {
16            return 0;
17        }
18
19        rev = rev * 10 + digit;
20    }
21
22    return rev * sign;
23};

File: 7-ReverseInteger/7-ReverseInteger.cpp
// Last updated: 3/18/2026, 11:07:42 PM
1class Solution {
2public:
3    int reverse(int x) {
4        long long rev = 0;  // Use long long to detect overflow during calculation
5
6        while (x != 0) {
7            int digit = x % 10;
8            x /= 10;
9
10            rev = rev * 10 + digit;
11
12            // Check overflow: must stay within 32-bit signed integer range
13            if (rev > INT_MAX || rev < INT_MIN) {
14                return 0;
15            }
16        }
17
18        return (int)rev;
19    }
20};
21